tests/integration/test_issue_1949.py

```diff
@@ -0,0 +1,107 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+import os.path
+
+import pytest
+
+from pex.cli.testing import run_pex3
+from pex.compatibility import commonpath
+from pex.testing import IntegResults, built_wheel, run_pex_command
+from pex.typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from typing import Any, Iterator
+
+
+@pytest.fixture
+def wheel():
+    # type: () -> Iterator[str]
+    with built_wheel(name="not_boto", version="2.49.0a1") as whl:
+        yield whl
+
+
+def test_resolve_arbitrary_equality(
+    tmpdir,  # type: Any
+    wheel,  # type: str
+):
+    # type: (...) -> None
+
+    lock = os.path.join(str(tmpdir), "lock.json")
+    run_pex3(
+        "lock",
+        "create",
+        "--no-pypi",
+        "--find-links",
+        os.path.dirname(wheel),
+        "not_boto===2.49.0a1",
+        "--indent",
+        "2",
+        "-o",
+        lock,
+    ).assert_success()
+
+    pex_root = os.path.join(str(tmpdir), "pex_root")
+
+    def create_pex_from_lock(*additional_args):
+        # type: (*str) -> IntegResults
+        return run_pex_command(
+            args=[
+                "--pex-root",
+                pex_root,
+                "--runtime-pex-root",
+                pex_root,
+                "--lock",
+                lock,
+            ]
+            + list(additional_args)
+            + ["--", "-c", "import not_boto; print(not_boto.__file__)"]
+        )
+
+    result = create_pex_from_lock()

```

Comment:
This test and the next fail before the fix in the same way as reported in #1949.